test/src/test/java/jenkins/security/ResourceDomainTest.java

```diff
@@ -0,0 +1,281 @@
+        configuration.setUrl(resourceRoot);
+    }
+
+    @Test
+    public void secondDomainBasics() throws Exception {
+        JenkinsRule.WebClient webClient = j.createWebClient();
+
+        { // DBS directory listing is shown as always
+            Page page = webClient.goTo("userContent");
+            Assert.assertEquals("successful request", 200, page.getWebResponse().getStatusCode());
+            Assert.assertTrue("still on the original URL", page.getUrl().toString().contains("/userContent"));
+            Assert.assertTrue("web page", page.isHtmlPage());
+            Assert.assertTrue("complex web page", page.getWebResponse().getContentAsString().contains("javascript"));
+        }
+
+        String resourceResponseUrl;
+        { // DBS on primary domain forwards to second domain when trying to access a file URL
+            webClient.setRedirectEnabled(true);
+            Page page = webClient.goTo("userContent/readme.txt", "text/plain");
+            resourceResponseUrl = page.getUrl().toString();
+            Assert.assertEquals("resource response success", 200, page.getWebResponse().getStatusCode());
+            Assert.assertNull("no CSP headers", page.getWebResponse().getResponseHeaderValue("Content-Security-Policy"));
+            Assert.assertTrue("Served from resource domain", resourceResponseUrl.contains(RESOURCE_DOMAIN));
+            Assert.assertTrue("Served from resource action", resourceResponseUrl.contains("static-files"));
+        }
+
+        { // direct access to resource URL works
+            Page page = webClient.getPage(resourceResponseUrl);
+            resourceResponseUrl = page.getUrl().toString();
+            Assert.assertEquals("resource response success", 200, page.getWebResponse().getStatusCode());
+            Assert.assertNull("no CSP headers", page.getWebResponse().getResponseHeaderValue("Content-Security-Policy"));
+            Assert.assertTrue("Served from resource domain", resourceResponseUrl.contains(RESOURCE_DOMAIN));
+            Assert.assertTrue("Served from resource action", resourceResponseUrl.contains("static-files"));
+        }
+
+        { // show directory index
+            webClient.setRedirectEnabled(false);
+            webClient.setThrowExceptionOnFailingStatusCode(false);
+            Page page = webClient.getPage(resourceResponseUrl.replace("readme.txt", ""));
+            Assert.assertEquals("directory listing response", 200, page.getWebResponse().getStatusCode());
+            String responseContent = page.getWebResponse().getContentAsString();
+            Assert.assertTrue("directory listing shown", responseContent.contains("readme.txt"));
+            Assert.assertTrue("is HTML", responseContent.contains("href="));
+        }
+
+        String resourceRootUrl = ExtensionList.lookupSingleton(ResourceDomainConfiguration.class).getUrl();
+        {
+            webClient.setThrowExceptionOnFailingStatusCode(false);
```

Comment:
You are not resetting this flag, so the subsequent sets are redundant, and you might be masking failing codes in later sections where it is not being explicitly set. Maybe you wanted to create a new client per block?

Reply:
It's set in every block except the last one, and there I assert HTTP 200.